src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -286,8 +286,8 @@ public static Exception LogExceptionMessage(EventLevel eventLevel, Exception exc
             if (IdentityModelEventSource.Logger.IsEnabled() && IdentityModelEventSource.Logger.LogLevel >= eventLevel)
                 IdentityModelEventSource.Logger.Write(eventLevel, exception.InnerException, exception.Message);
 
-            if (Logger.IsEnabled(eventLevel))
-                Logger.Log(WriteEntry(eventLevel, exception.InnerException, exception.Message, null));
+            if (Logger.IsEnabled((EventLogLevel)eventLevel))
+                Logger.Log(WriteEntry((EventLogLevel)eventLevel, exception.InnerException, exception.Message, null));
 
```

Comment:
Performing a direct explicit cast without checks. Let me know if Enum.TryParse(..) or Enum.IsDefinied(..) checks are preferable to make sure a mapping exists.

Reply:
I don't think the int definition check would be costly and will give a much better devEx if anything ever goes wrong. Would be interesting to benchmark it though: [https://benchmarkdotnet.org/](https://benchmarkdotnet.org/)

Reply:
Agreed, I ran the benchmark and it takes 76ns on an average to evaluate the condition if the IsDefined check is in place. I'll add the check and then run the performance tests between the dev and this topic branch to compare the performance impact.